app.js

```diff
@@ -1,130 +1,178 @@
-  renderSetFunction(setArray);
+/* INVENTORY
+  I have: 
+          -connected the value of radio buttons to the id           of the prouct (this value means I can access the id of the product upon productSelectionCount
+          -Names of products stored in var
+          -src of images stored in variable
+          - a function that returns three randomly generated products
+          - a function that generates individual random products
+          - declared state variables
+          - a function that sets an intial state
+  */
 
 
 
+// 3. need an eventListener that will 
+    // call the display function()
+    // must have be able to increment the values of state variables: productSelectionCount ; total views ; total sets
+    // must discern between rendering new content OR ending survery
+        // conditions are if sets >= 25, survery ends and will DISPLAY RESULTS. If not, render new content and continue to increment.
 
+const form = document.querySelector('form')
+//this is creating a variable that is assigned to the value of the form and its contents
 
-  const userChoice = document.querySelector('input:checked')
-  click++ 
-});
 
+form.addEventListener("submit", (e) => {
+    e.preventDefault();
+    console.log('go');
+    
+    // conditions the circumsances of survey rendering
+    if (totalSets >= 25) {
+        displayResults()
+    }
 
-export const getValidProducts = (products, priorSet) => {
-  for (let i = 0; i < length.priorSet; i++) {
-      let i = priorSet[i];
-      let validProducts = products.splice(priorSet(i)); 
-  }
-  return validProducts;
-};
+    const formData = new FormData(form);
+    //this is taking in the form var, so the value of formData(var) is the data of the form itself upon event.
 
-export const setArrayFunction = (validProducts) => {
-    let setArray = []
-    let itemOne = Math.floor(Math.random() * validProducts.length);
-    setArray.push(itemOne.image);
-    let itemTwo = Math.floor(Math.random() * validProducts.length);
-    setArray.push(itemTwo.image);
-    let itemThree = Math.floor(Math.random() * validProducts.length);
-    setArray.push(itemThree.image)
+    const selectedProductId = formData.get('selected');
+    //this is assigning the value of the selected product to the var
 
-    return setArray;
-};
+    totalSets++;
+    //incrementing 
+    
+    //here checkcking whether or not the selected product already exists in the array passed as first argument.
+        //this works because the value of the button is assigned to the id of the product
+    const productExistCheck = findById(productSelectionCount, selectedProductId);
 
-export placePhotos {
-    if clicks < 25 {
-      // render set array on page
+    //conditional to either push that product into array as new object or add it to existing object
+    if (productExistCheck) {
+        productExistCheck.votes++;
     }
-}
+
+    productSelectionCount.push({
+        id: selectedProductId,
+        votes: 1,
+    })   
```

Comment:
This should be in an else block unless you want a new object being pushed into the array every time the function runs.